client/app/containers/index.js

```diff
@@ -41,21 +45,41 @@ class App extends React.Component {
   }
 
   onFileClick(link) {
-    const newPath = `${this.state.path}/${link}`;
+    this.getFile(this.state.path, link).then(file => this.setState(file));
+  }
+
+  getFile(path, title) {
+    const newPath = `${path}/${title}`;
 
     const handleResponse = (response) => {
-      this.setState({
+      return {
         currentFile: {
-          title: link,
+          title,
           __html: response.content
         }
-      });
+      };
     };
 
-    api.getFile(newPath)
+    return api.getFile(newPath)
       .then(handleResponse, this.handleError);
   }
 
+  getRootReadme(newState) {
```

Comment:
If so, then, imho, we probably should name fields in server response similar to react state.

Like so:
getDirectoryContent
```
{
   directoryEntry: [],
}
```

getFileContent
```
{
   currentFile: {
      title: '',
      __html: ''
    }
}
```
This will allow us just merge new state right from response, without additional validation.
And if root readme is available - client will receive it just like usual state "snapshot".
What you think?